src/main/java/com/google/sps/servlets/EventServlet.java

```
@@ -65,7 +73,54 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
     @Override
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        // ToDo: Mingyi implment this method.
-        response.getWriter().println("You've called the post request");
+        String jsonBody = new BufferedReader(new InputStreamReader(request.getInputStream())).lines().collect(
+            Collectors.joining("\n"));
+
+        // Error handling: return error if jsonBody is empty
+        if (jsonBody == null || jsonBody.trim().length() == 0) {
+            return;
```

This should throw an error rather than silently fail